### ISSUE TITLE
Error Calling Amplicon Variants with Nanopore

### ISSUE THREAD

**Issue by laksyn** [jts/nanopolish#320]
Thanks for developing Nanopolish and for your help.  I’m trying to use Nanopolish to call variants from MinION reads of amplicons (~2Kb long; ~ 50,000 reads each) and am getting an error.  Here is what I have done:

1. Albacore – to base call and separate barcodes
   >read_fast5_basecaller.py --flowcell FLO-MIN106 --kit SQK-LSK108 --barcoding --recursive     
     --output_format fast5,fastq --input /path/to/fast5 --save_path /path/to/fast5_basecalled 
     –worker_threads 4
2. Concatinated all fastq files into one file
3. Minimap2 – aligned reads to reference
    > minimap2-2.7_x64-linux/minimap2 -ax map-ont /path/to/Barcode02_ref.fa /path/to/BC02.fasta 
          > aln_barcode02.sam
4.Samtools to convert sam to bam and sort bam
5. Nanopolish – attempt to call variants
    >  nanopolish variants  --reads /path/to/All_barcode02.fastq 
        --bam /path/to/aln_barcode02.sorted.bam --genome /path/to/Barcode02_ref.fa 
        --outfile Barcode02.vcf
Produces Error:
TODO: train model
TODO: filter data
Error: could not find path to FAST5 for read 0001c491-59ea-43c4-a23f-7d9831259087
Please make sure that this path is accessible: barcode=barcode02

When I looked at the Nanopore issues page for this error, it seemed like this might mean nanopolish couldn’t find the path to the fasta5 files and that perhaps indexing would be needed.  So I tried running the indexing command:

   > nanopolish index -d /path/to/directory/with/*.fast5 reads.fastq

To do this, I copied all the *.fast5 files into a single directory (kept getting errors when I tried to use multiple -d).  This command runs without errors or messages but no index files are produced (fai, gzi, readdb).  I read in some of the posted issues that the fastq file may need to include the path to the fast5 in the header.  I don’t see this in the Albacore fastq files but I don’t know what I might need to do to get Albacore to include this.  Is there an option I should have specified in the command line to get this?  Any suggestions you have would be appreciated.  I’ve experimented with a lot of things without success.

**Comment by jts** [comment]
Hi,

`nanopolish index` is recursive. You should just have to run:

```
nanopolish index -d /path/to/fast5 BC02.fasta
```

Jared

**Comment by laksyn** [comment]
Hi Jared,
The command you suggested was the first thing I tried but it didn't work.  The prompt returns almost instantaneously, there is no feedback messages or error messages, and no index files (index, fai, gzi, readdb).  It was only after that command failed that I tried to do some weird stuff like copying all the Fast5 files into the same directory.    

It is almost like the nanopolish index command isn't even running.  When I run "nanopolish index -h", I don't even get a help menu (like I do if I type "nanopolish variaints -h" or "nanopolish eventalign -h".

**Comment by jts** [comment]
Can you confirm you are using the very latest version from github?

**Comment by laksyn** [comment]
I just installed nanopolish from github last week (https://github.com/jts/nanopolish).

**Comment by laksyn** [comment]
Jared -- can you offer any further advice?  Do you think there may be something wrong with my installation?  Since I only downloaded the program last week, I think it should be the latest.  When I request version I get --  Version 0.4.0

**Comment by jts** [comment]
Sorry I've been tied up the last few days so I haven't been able to answer many issues.

That is very old, it should be 0.8.5. Perhaps you have an old version on your `PATH` that is being erroneously run instead of the version you just installed?

**Comment by jts** [comment]
Any luck updating?

**Comment by laksyn** [comment]
Hi Jared,
Sorry, I too got tied up with other tasks.  Now I'm back to the fun stuff.

Many thanks for your insight on the version because it turned out I was inadvertently running an old version of nanopolish.  I had installed Bio-Linux originally and then installed Nanopolish from github.  I think there may have been an old version of Nanopolish pre-installed with Bio-Linux(?).  Anyway, I corrected that problem and am now running Version 0.8.5.  The indexing ran and generated the expected files (.index, .fai, .gzi.readdb).  That's great! 

Many thanks for taking the time to help me.
